### PR TITLE
Push dependency on the GCECloud up out of the neg controller

### PR DESCRIPTION
**Pull request by bowei**

### CODE REVIEW

File: pkg/neg/cloudprovideradapter.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/*
```
Comment (rramkumar1) [inline pkg/neg/cloudprovideradapter.go]: Were your planning on using this adapter in a follow up?

Comment (bowei) [reply pkg/neg/cloudprovideradapter.go]: it's used...

Comment (rramkumar1) [reply pkg/neg/cloudprovideradapter.go]: I don't see NewAdapter() called anywhere. Maybe I am missing it...

Comment (bowei) [reply pkg/neg/cloudprovideradapter.go]: good catch, added, it got lost in a merge

**Pull request merged**